dialogy/plugins/text/classification/xlmr.py

```diff
@@ -74,61 +66,94 @@ def __init__(
         self.model_dir = self.args_map.get("best_model_dir")
-
         if not self.model_dir:
             raise ValueError(
                 f"'best_model_dir' missing in passed args_map."
             )
-
-        self.use_calibration = self.args_map.get(const.MODEL_CALIBRATION, False)
-
-        self.labelencoder_file_path = os.path.join(
-            self.model_dir, const.LABELENCODER_FILE
-        )
+        
         self.ts_parameter: float = (
             read_from_json(
                 [const.TS_PARAMETER], self.model_dir, const.CALIBRATION_CONFIG_FILE
             ).get(const.TS_PARAMETER)
             or 1.0
         )
 
-        self.threshold = threshold
-        self.skip_labels = set(skip_labels or set())
-
-        self.round = score_round_off
+        # flag that specifies whether plugin is being imported externally solely for model
+        imported = kwargs.get("imported", False)
 
-        self.kwargs = kwargs or {}
+        if self.purpose == const.TRAIN or imported:
+            self.use_cuda = self.purpose == const.TRAIN
+            try:
+                classifer = getattr(
+                    importlib.import_module(const.XLMR_MODULE), const.XLMR_MULTI_CLASS_MODEL
+                )
+            except ModuleNotFoundError as error:
+                raise ModuleNotFoundError(
+                    "Plugin requires simpletransformers -- https://simpletransformers.ai/docs/installation/"
+                ) from error
+            
+            self.labelencoder = preprocessing.LabelEncoder()
+            self.classifier = classifer
+            self.model: Any = None
+
+            self.labelencoder_file_path = os.path.join(
+                self.model_dir, const.LABELENCODER_FILE
+            )
 
-        # TODO: check if this can be avoided
-        del self.kwargs["name"]
+            self.kwargs = kwargs or {}
+
+            # TODO: check if this can be avoided
+            if "name" in self.kwargs:
+                del self.kwargs["name"]
+            if "imported" in self.kwargs:
+                del self.kwargs["imported"]
+
+            try:
+                if os.path.exists(self.labelencoder_file_path):
+                    self.init_model()
+            except EOFError:
+                logger.error(
+                    f"Plugin {self} Failed to load labelencoder from {self.labelencoder_file_path}. "
+                    "Ignore this message if you are training but if you are using this in production or testing, then this is serious!"
+                )
 
-        try:
-            if os.path.exists(self.labelencoder_file_path):
-                self.init_model()
-        except EOFError:
-            logger.error(
-                f"Plugin {self} Failed to load labelencoder from {self.labelencoder_file_path}. "
-                "Ignore this message if you are training but if you are using this in production or testing, then this is serious!"
+        elif self.purpose == const.PRODUCTION:
```

Comment:
model eval can also happen with model served through model inference service (locally, if needed). This will make sure that inference serving exactly matches inference during evaluation.

```suggestion
        elif self.purpose in [const.PRODUCTION, const.TEST]:
```

Reply:
Removing this.. better to keep eval within core slu service to minimize the changes needed inside retraining pipeline.